src/write_stage.rs

```diff
@@ -76,25 +147,162 @@ impl WriteStage {
                             Error::RecvTimeoutError(RecvTimeoutError::Disconnected) => break,
                             Error::RecvTimeoutError(RecvTimeoutError::Timeout) => (),
                             _ => {
-                                inc_new_counter!("write_stage-error", 1);
+                                inc_new_counter!("write_stage-write_and_send_entries-error", 1);
                                 error!("{:?}", e);
                             }
                         }
                     };
+                    if let Err(e) = Self::leader_vote(
+                        debug_id,
+                        &keypair,
+                        &bank,
+                        &crdt,
+                        &blob_recycler,
+                        &vote_blob_sender,
+                        &mut last_vote,
+                    ) {
+                        inc_new_counter!("write_stage-leader_vote-error", 1);
+                        error!("{:?}", e);
+                    }
                 }
             })
             .unwrap();
 
-        (WriteStage { thread_hdl }, blob_receiver)
+        let thread_hdls = vec![t_responder, thread_hdl];
+        (WriteStage { thread_hdls }, blob_receiver)
     }
 }
 
 impl Service for WriteStage {
     fn thread_hdls(self) -> Vec<JoinHandle<()>> {
-        vec![self.thread_hdl]
+        self.thread_hdls
     }
-
     fn join(self) -> thread::Result<()> {
-        self.thread_hdl.join()
+        for thread_hdl in self.thread_hdls() {
+            thread_hdl.join()?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use bank::Bank;
+    use crdt::{Crdt, NodeInfo};
```

Comment:
Can this unit test be written without a crdt dependency?

Reply:
Don't believe so, `leader_vote` takes a crdt object:

```
pub fn leader_vote(
    debug_id: u64,
    keypair: &KeyPair,
    bank: &Arc<Bank>,
    crdt: &Arc<RwLock<Crdt>>,   <===
    blob_recycler: &BlobRecycler,
    vote_blob_sender: &BlobSender,
    last_vote: &mut u64,
) -> Result<()> {
```